[Bug] Docker build fails on migrator

Image construction fails on the main branch.

```
% make build-docker-dev
build development container
Info: the frontend code is expected to be built already.
go run build.go -goos linux -pkg-arch amd64  build latest
Version: 8.3.0, Linux Version: 8.3.0, Package Iteration: 1637161033pre
building binaries build
building grafana-server ./pkg/cmd/grafana-server
rm -r ./bin/linux-amd64/grafana-server
rm -r ./bin/linux-amd64/grafana-server.md5
go build -ldflags -w -X main.version=8.3.0-pre -X main.commit=b233a7d8d2 -X main.buildstamp=1637075007 -X main.buildBranch=main -o ./bin/linux-amd64/grafana-server ./pkg/cmd/grafana-server
go version
go version go1.17.2 darwin/amd64
Targeting linux/amd64
building binaries build
building grafana-cli ./pkg/cmd/grafana-cli
rm -r ./bin/linux-amd64/grafana-cli
rm -r ./bin/linux-amd64/grafana-cli.md5
go build -ldflags -w -X main.version=8.3.0-pre -X main.commit=b233a7d8d2 -X main.buildstamp=1637075007 -X main.buildBranch=main -o ./bin/linux-amd64/grafana-cli ./pkg/cmd/grafana-cli
# github.com/grafana/grafana/pkg/services/sqlstore/migrator
pkg/services/sqlstore/migrator/sqlite_dialect.go:127:16: undefined: sqlite3.Error
pkg/services/sqlstore/migrator/sqlite_dialect.go:138:16: undefined: sqlite3.Error
pkg/services/sqlstore/migrator/sqlite_dialect.go:146:33: undefined: sqlite3.ErrConstraintUnique
exit status 2
exit status 1
make: *** [build-docker-dev] Error 1
```

A `make all` passes just fine. The migrator code is also correct.

```
% cd pkg/services/sqlstore/migrator/
% go build
%
```

Thanks @xlson. The `build-docker-full` also fails.

```
% make build-docker-full
[…]
 => [go-builder 12/14] COPY .bingo .bingo                                                            0.0s
 => CANCELED [go-builder 13/14] RUN go mod verify                                                  322.6s
 => [js-builder  8/13] COPY tsconfig.json .eslintrc .editorconfig .browserslistrc .prettierrc.js ba  0.1s
 => [js-builder  9/13] COPY public public                                                            2.5s
 => [js-builder 10/13] COPY tools tools                                                              0.0s
 => [js-builder 11/13] COPY scripts scripts                                                          0.1s
 => [js-builder 12/13] COPY emails emails                                                            0.1s
 => ERROR [js-builder 13/13] RUN yarn build                                                        259.9s
------
 > [js-builder 13/13] RUN yarn build:
------
executor failed running [/bin/sh -c yarn build]: exit code: 129
make: *** [build-docker-full] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -7,7 +7,7 @@ WIRE_TAGS = "oss"
 -include local/Makefile
 include .bingo/Variables.mk
 
-.PHONY: all deps-go deps-js deps build-go build-server build-cli build-js build build-docker-dev build-docker-full lint-go golangci-lint test-go test-js gen-ts test run run-frontend clean devenv devenv-down protobuf drone help
+.PHONY: all deps-go deps-js deps build-go build-server build-cli build-js build build-docker-full lint-go golangci-lint test-go test-js gen-ts test run run-frontend clean devenv devenv-down protobuf drone help
 
 GO = go
 GO_FILES ?= ./pkg/...
@@ -94,13 +94,6 @@ shellcheck: $(SH_FILES) ## Run checks for shell scripts.
 	$(SH_FILES) -e SC1071 -e SC2162
 
 ##@ Docker
-
-build-docker-dev: ## Build Docker image for development (fast).
-	@echo "build development container"
-	@echo "\033[92mInfo:\033[0m the frontend code is expected to be built already."
-	$(GO) run build.go -goos linux -pkg-arch amd64 ${OPT} build latest
-	cp dist/grafana-latest.linux-x64.tar.gz packaging/docker
-	cd packaging/docker && docker build --tag grafana/grafana:dev .
 
 build-docker-full: ## Build Docker image for development.
 	@echo "build docker container"

--- a/contribute/developer-guide.md
+++ b/contribute/developer-guide.md
@@ -204,11 +204,6 @@ make build-docker-full
 
 The resulting image will be tagged as grafana/grafana:dev.
 
-> **Note:** If you've already set up a local development environment, and you're running a `linux/amd64` machine, you can speed up building the Docker image:
-
-1. Build the frontend: `go run build.go build-frontend`.
-1. Build the Docker image: `make build-docker-dev`.
-
 **Note:** If you are using Docker for macOS, be sure to set the memory limit to be larger than 2 GiB. Otherwise, `grunt build` may fail. The memory limit settings are available under **Docker Desktop** -> **Preferences** -> **Advanced**.
 
 ## Troubleshooting

--- a/pkg/build/cmd.go
+++ b/pkg/build/cmd.go
@@ -5,10 +5,8 @@ import (
 	"flag"
 	"fmt"
 	"go/build"
-	"io/ioutil"
 	"log"
 	"os"
-	"path"
 	"path/filepath"
 	"strconv"
 	"strings"
@@ -102,9 +100,6 @@ func RunCmd() int {
 				return logError("error packaging dist directory", err)
 			}
 
-		case "latest":
-			makeLatestDistCopies()
-
 		case "clean":
 			clean(opts)
 
@@ -115,35 +110,6 @@ func RunCmd() int {
 	}
 
 	return 0
-}
-
-func makeLatestDistCopies() {
-	files, err := ioutil.ReadDir("dist")
-	if err != nil {
-		log.Fatalf("failed to create latest copies. Cannot read from /dist")
-	}
-
-	latestMapping := map[string]string{
-		"_amd64.deb":               "dist/grafana_latest_amd64.deb",
-		".x86_64.rpm":              "dist/grafana-latest-1.x86_64.rpm",
-		".linux-amd64.tar.gz":      "dist/grafana-latest.linux-x64.tar.gz",
-		".linux-amd64-musl.tar.gz": "dist/grafana-latest.linux-x64-musl.tar.gz",
-		".linux-armv7.tar.gz":      "dist/grafana-latest.linux-armv7.tar.gz",
-		".linux-armv7-musl.tar.gz": "dist/grafana-latest.linux-armv7-musl.tar.gz",
-		".linux-armv6.tar.gz":      "dist/grafana-latest.linux-armv6.tar.gz",
-		".linux-arm64.tar.gz":      "dist/grafana-latest.linux-arm64.tar.gz",
-		".linux-arm64-musl.tar.gz": "dist/grafana-latest.linux-arm64-musl.tar.gz",
-	}
-
-	for _, file := range files {
-		for extension, fullName := range latestMapping {
-			if strings.HasSuffix(file.Name(), extension) {
-				if _, err := runError("cp", path.Join("dist", file.Name()), fullName); err != nil {
-					log.Println("error running cp command:", err)
-				}
-			}
-		}
-	}
 }
 
 func yarn(params ...string) {

